hydra-cluster/src/CardanoNode.hs

```diff
@@ -259,18 +259,22 @@ withCardanoNode ::
   NetworkId ->
   FilePath ->
   CardanoNodeArgs ->
-  (RunningNode -> IO ()) ->
-  IO ()
+  (RunningNode -> IO a) ->
+  IO a
 withCardanoNode tr networkId stateDirectory args@CardanoNodeArgs{nodeSocket} action = do
   traceWith tr $ MsgNodeCmdSpec (show $ cmdspec process)
   traceWith tr $ MsgNodeStarting{stateDirectory}
   withLogFile logFilePath $ \out -> do
     hSetBuffering out NoBuffering
     withCreateProcess process{std_out = UseHandle out, std_err = UseHandle out} $
       \_stdin _stdout _stderr processHandle ->
-        race_
-          (checkProcessHasNotDied "cardano-node" processHandle)
-          waitForNode
+        ( race
+            (checkProcessHasNotDied "cardano-node" processHandle)
+            waitForNode
+            >>= \case
+              Left{} -> error "should never been reached"
+              Right a -> pure a
```

Comment:
Can you explain what's the benefit of having `a` in the return type?

Reply:
It's needed because this function appears within a stack of `withXX action` functions where we need the return value of the action

Reply:
I found my answer: because you want to return the `Summary` out of the benchmark and this `a` is bubbling up, forcing you to do use `race` instead of `race_` and hence absurd handling of `Left ()` which will never be reached.